// Copyright 2016 Intel Corporation
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
// -----------------------------------------------------------------------------

syntax = "proto3";
option java_multiple_files = true;
option java_package = "sawtooth.sdk.protobuf";

import "batch.proto";


message Block {
    int32 block_num = 1;
    string previous_block_id = 2;
    repeated Batch batches = 3;
    map<string,string> consensus= 4;
    string state_root_hash = 5;
    string signature = 6;
    string id = 7;
}

message BlockState {
    Block block = 1;
    int32 weight = 2;
    BlockStatus status = 3;
}

enum BlockStatus{
    Unknown = 0;
    Invalid = 1;
    Valid = 2;
}
